api/admin/adminRouter.js

```diff
@@ -14,7 +14,7 @@ router.get('/', authRequired, function (req, res) {
       res.status(500).json({ message: err.message });
     });
 });
-router.get('/:id', checkAdminExist, authRequired, function (req, res, next) {
+router.get('/:id', authRequired, checkAdminExist, function (req, res, next) {
```

Comment:
This was a really good catch, and an important item to fix - thanks for getting us closer to solid authentication and security!